src/core/vectortile/qgsvectortilelayerrenderer.cpp

```diff
@@ -163,8 +163,8 @@ bool QgsVectorTileLayerRenderer::render()
       ctx.labelingEngine()->removeProvider( mLabelProvider );
       mLabelProvider = nullptr; // provider is deleted by the engine
     }
-
-    mRequiredLayers.unite( mLabelProvider->requiredLayers( ctx, mTileZoom ) );
+    else
```

Comment:
Was this an accident? Or unrelated fix?

Reply:
No, there is nullptr acces when you passed into the above if: `mLabelProvider = nullptr` then the pointer is dereferenced at line 167.

Reply:
Can this be moved to a separate PR please? then we can backport if necessary